### PR TITLE
PR to merge Feature branch to Develop branch

### PR DESCRIPTION
**Pull request by Haroonabdulrazaq**
- Add Sign in and Sign up
- Add Events model and Controller
- Add User model and Controller
- Add through table (Event Attendance)
- Add navigation
- Use sessions to Login User
- Add Registration for an Event
- Add necessary Views

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @Haroonabdulrazaq and @luciano-ilha 
### You have done well so far :clap: However, you have a few issues you need to fix:-
- [x] A user is unable to create events
- [x] User model use correct validation, see below code
`validates :name, presence: true`
- [x] Remove the comments in your codebase
### Make the changes and submit for another code review. Good luck!
- Happy coding!

By Raymond
Slack @iamraymond

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  def create
+    @event = current_user.events.build(event_params)
+    if @event.save
+      flash[:notice] = 'Event Created Successfully'
+      redirect_to root_path
+    else
+      flash.now[:alert] = 'Form not submitting for some reasons'
+      render :new
+    end
+  end
```
Comment (rOluochKe) [inline app/controllers/events_controller.rb:20]: ### Pass in event_params method as private and list all the attributes
```
  def event_params
    params.require(:event).permit(:attr1, :attr2)
  end
```

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class EventsController < ApplicationController
+  def index
+    @events = Event.all
+  end
```
Comment (rOluochKe) [inline app/controllers/events_controller.rb]: ### When listing events, you can do something like this:-
```
@past_events = Event.previous_events
@upcoming_events = Event.upcoming_events
```

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      redirect_to root_path
+    else
+      flash.now[:alert] = 'Form not submitting for some reasons'
+      render :new
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
```
Comment (rOluochKe) [inline app/controllers/events_controller.rb:24]: ### You can list events based on the following criteria:-
lists events created by user `@created_events = current_user.events`
lists events previoulsy attendented by user `@past_events = current_user.attended_events.previous`
lists upcoming events previously attended by user `@upcoming_events = current_user.attended_events.upcoming`

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (Haroonabdulrazaq) [reply app/controllers/events_controller.rb:20]: They are all listed in the application_controller, we felt its keeps our controller clean and neat

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team!
### You have made good progress so far, however, there are still issues you need to address:-
- [x] It is not clear how one can attend the event? Make the correction
- [x] The event's show page displays the list of attendants
- [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming
### Make these changes and submit for another code review. Good luck!
- Happy hacking!

**Review by rOluochKe — CHANGES_REQUESTED** [review]
### You are almost there, but I still get some error when I try to register for the event, refer from image below
eror
### Make the change and submit for another code review. Good luck.
- Happy coding!

**Review by rOluochKe — APPROVED** [review]
### You finally got it right,  the project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

### [Improvements]
- You can do a better design to present the project.

**Pull request merged**